plasmapy/analysis/swept_langmuir/ion_saturation_current.py

```diff
@@ -0,0 +1,228 @@
+    current_bound: float = None,
+    voltage_bound: float = None,
+) -> Tuple[ffuncs.Linear, ISatExtras]:
+    """
+    Determines the ion-saturation current (:math:`I_{sat}`) for a given
+    current-voltage (IV) curve obtained from a swept Langmuir probe.
+    The current collected by a Langmuir probe reaches ion-saturation
+    when the probe is sufficiently biased so the influx of electrons is
+    completely repelled leading to only the collection of ions.  (For
+    additional details see the **Notes** section below.)
+
+    **Aliases:** `find_isat_`
+
+    Parameters
+    ----------
+
+    voltage: `numpy.ndarray`
+        1-D numpy array of monotonically increasing probe biases
+        (should be in volts).
+
+    current: `numpy.ndarray`
+        1-D numpy array of probe current (should be in amperes)
+        corresponding to the ``voltage`` array.
+
+    fit_type: `str`
+        The type of curve (:term:`fit-function`) to be fitted to the
+        Langmuir trace, valid options are listed below.
+        (DEFAULT ``"exp_plus_linear"``)
+
+        +-----------------------+----------------------------------------------------------+
+        | ``"linear"``          | `~plasmapy.analysis.fit_functions.Linear`                |
+        +-----------------------+----------------------------------------------------------+
+        | ``"exponential"``     | `~plasmapy.analysis.fit_functions.ExponentialPlusOffset` |
+        +-----------------------+----------------------------------------------------------+
+        | ``"exp_plus_linear"`` | `~plasmapy.analysis.fit_functions.ExponentialPlusLinear` |
+        +-----------------------+----------------------------------------------------------+
+
+    current_bound: `float`
+        A fraction representing a percentile window around the minimum
+        current for which to collect the points.  For example, a value
+        of ``0.1`` indicates to use all points within 10% of the
+        minimum current.  (DEFAULT ``None``)
+
+        |
+
+        If neither ``current_bound`` or ``voltage_bound`` are specified,
+        then the routine will collect indices based on an internal
+        ``current_bound`` setting for the specified ``fit_type``.
+
+        +-----------------------+--------------------------------------+
+        | ``"linear"``          | 0.4                                  |
+        +-----------------------+--------------------------------------+
+        | ``"exponential"``     | 1.0                                  |
+        +-----------------------+--------------------------------------+
+        | ``"exp_plus_linear"`` | 1.0                                  |
+        +-----------------------+--------------------------------------+
+
+    voltage_bound: `float`
+        A bias voltage (in volts) that specifies an upper bound used to
+        collect the points for the curve fit.  That is, points that
+        satisfy ``voltage <= voltage_bound`` are used in the fit.
```

Comment:
Sooooooo tempted to suggest a unicode character here even though it would be a bad idea!